crates/sui-core/src/transaction_input_checker.rs

```diff
@@ -17,12 +18,94 @@ use tracing::{debug, instrument};
 
 use crate::authority::SuiDataStore;
 
+pub struct InputObjects {
+    objects: Vec<(InputObjectKind, Object)>,
+}
+
+impl InputObjects {
+    pub fn new(objects: Vec<(InputObjectKind, Object)>) -> Self {
+        Self { objects }
+    }
+
+    pub fn len(&self) -> usize {
+        self.objects.len()
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.objects.is_empty()
+    }
+
+    pub fn filter_owned_objects(&self) -> Vec<ObjectRef> {
+        let owned_objects: Vec<_> = self
+            .objects
+            .iter()
+            .filter_map(|(object_kind, object)| match object_kind {
+                InputObjectKind::MovePackage(_) => None,
+                InputObjectKind::ImmOrOwnedMoveObject(object_ref) => {
+                    if object.is_immutable() {
+                        None
+                    } else {
+                        Some(*object_ref)
+                    }
+                }
+                InputObjectKind::SharedMoveObject(_) => None,
+            })
+            .collect();
+
+        debug!(
+            num_mutable_objects = owned_objects.len(),
```

Comment:
This looks left over from elsewhere--I don't see where `num_mutable_objects` is bound.

Reply:
Ha, so this is confusing but `debug!` is a tracing macro that dumps a debug log. The syntax `num_mutable_objects = owned_objects.len()` adds an indexable metric/key to the log named `num_mutable_objects` with the value of `owned_objects.len()`